source

```diff
@@ -89176,6 +89190,121 @@ interface <dfn interface>BeforeUnloadEvent</dfn> : <span>Event</span> {
 
   </div>
 
+  <h4>The <code>NotRestoredReasons</code> interface</h4>
+
+  <pre><code class="idl">enum <dfn enum>PreventedBackForwardCache</dfn> { "yes", "no", "masked" };
+  [Exposed=Window]
+  interface <dfn interface>NotRestoredReasons</dfn> {
+    readonly attribute <span>PreventedBackForwardCache</span> <span
+  data-x="dom-not-restored-reasons-prevented-back-forward-cache">preventedBackForwardCache</span>;
+    readonly attribute DOMString? <span data-x="dom-not-restored-reasons-src">src</span>;
+    readonly attribute DOMString? <span data-x="dom-not-restored-reasons-id">id</span>;
+    readonly attribute DOMString? <span data-x="dom-not-restored-reasons-name">name</span>;
+    readonly attribute DOMString? <span data-x="dom-not-restored-reasons-url">url</span>;
+    readonly attribute ObservableArray&lt;DOMString&gt;? <span data-x=
+     "dom-not-restored-reasons-reasons">reasons</span>;
+    readonly attribute ObservableArray&lt;<span>NotRestoredReasons</span>&gt;? <span
+     data-x="dom-not-restored-reasons-children">children</span>;
+  };</code></pre>
+
+  <dl class="domintro">
+    <dt><code data-x=""><var>notRestoredReasons</var>.<span subdfn data-x=
+      "dom-not-restored-reasons-prevented-back-forward-cache">preventedBackForwardCache</span>
+    </code></dt>
+    <dd><p>Returns whether the document has blocked back/forward cache, from the list below.</p></dd>
+    <dt><code data-x=""><var>notRestoredReasons</var>.<span subdfn data-x=
+      "dom-not-restored-reasons-src">src</span></code></dt>
+      <dd><p>Returns the src attribute of the document's iframe if the document is in an iframe.
```

Comment:
We need to formally say that something is in an iframe. Maybe "in a child navigable" or something? See also https://html.spec.whatwg.org/multipage/document-sequences.html#child-navigable

Similarly when referring to "the document's iframe", maybe use "the document's content navigable", and the origin checks below too. In general we need to use precise spec terms outside of notes, since these are not comments but instructions.

Also the reference to the src attribute should probably be linked too. What I normally do to find these things is I look for "iframe element src mdn" or something like that, and that the MDN usually has a link to the relevant part of the HTML spec.

Reply:
Thanks! I updated the expressions to be more formally.